src/svc/roles.cpp

```diff
@@ -1,6 +1,43 @@
 #include "roles.h"
 
 namespace svc {
+grpc::ServerUnaryReactor *Roles::AddPermission(
+	grpc::CallbackServerContext *context, const gk::v1::RolesAddPermissionRequest *request,
+	gk::v1::Role *response) {
+	auto *reactor = context->DefaultReactor();
+
+	// TODO: error handling
+	auto role        = datastore::RetrieveRole(request->role_id());
+	auto policyAttrs = datastore::ListRbacPolicyAttrsByRole(role.id());
+
+	auto permissionId = request->id();
+	role.addPermission(permissionId);
+
+	for (const auto &row : policyAttrs) {
```

Comment:
This is ok but you could also use C++17 structured bindings here to make it look a bit "cleaner". e.g. 
```suggestion
	for (const auto &[policyId, attrs] : policyAttrs) {
```